src/wifi.rs

```diff
@@ -1235,6 +1333,36 @@ impl<'d> Drop for EspWifi<'d> {
 #[cfg(esp_idf_comp_esp_netif_enabled)]
 unsafe impl<'d> Send for EspWifi<'d> {}
 
+#[cfg(esp_idf_comp_esp_netif_enabled)]
```

Comment:
why is this implemented two times? onetime behind this feature flag and first time without on on line 1020

Reply:
Because both the nw level 2 `WifiDriver` as well as the ip nw level 3 `EspWifi` thing can be wrapped in `AsyncWifi` or `BlockingWifi`? Basically, if the user does not need the nw level 3 layer (ip) he can still consume the nw level 2 with the async / blocking wrappers.